lldb/source/Plugins/Language/Swift/SwiftLanguage.cpp

```diff
@@ -1763,95 +1764,104 @@ bool SwiftLanguage::GetFunctionDisplayName(
       s << display_name;
       return true;
     }
-    const char *cstr = display_name.data();
-    const char *open_paren = strchr(cstr, '(');
-    const char *close_paren = nullptr;
-    const char *generic = strchr(cstr, '<');
-    // If before the arguments list begins there is a template sign
-    // then scan to the end of the generic args before you try to find
-    // the arguments list.
-    if (generic && open_paren && generic < open_paren) {
-      int generic_depth = 1;
-      ++generic;
-      for (; *generic && generic_depth > 0; generic++) {
-        if (*generic == '<')
-          generic_depth++;
-        if (*generic == '>')
-          generic_depth--;
-      }
-      if (*generic)
-        open_paren = strchr(generic, '(');
-      else
-        open_paren = nullptr;
-    }
-    if (open_paren) {
-      close_paren = strchr(open_paren, ')');
+    return GetFunctionDisplayArgs(args, s, display_name, exe_scope);
+  }
+  }
+  return false;
+}
+
+bool SwiftLanguage::GetFunctionDisplayArgs(
```

Comment:
Lets add this helper function in a separate patch. Makes it easier to review

Reply:
This method is needed for the implementation of `bool SwiftLanguage::HandleFrameFormatVariable`. I don't think I can remove it without copy pasting the logic at the call site.

Say this is PR A. Would it be OK to create a new PR B which introduces this helper function, merge PR B and then rebase the PR A on the changes introduced by PR B?

Reply:
> Say this is PR A. Would it be OK to create a new PR B which introduces this helper function, merge PR B and then rebase the PR A on the changes introduced by PR B?

Yup that's the way to go